tests/drivers/watchdog/wdt_error_cases/src/main.c

```diff
@@ -0,0 +1,1001 @@
+ */
+#define WDT_FLAG_RESET_NONE_IS_NOT_SUPPORTED
+#define WDT_FLAG_RESET_CPU_CORE_IS_NOT_SUPPORTED
+#define WDT_FLAG_RESET_SOC_IS_NOT_SUPPORTED
+#define WATCHDOG_SUPPORTS_ONLY_ONE_TIMEOUT_VALUE
+#define WDT_OPT_PAUSE_IN_SLEEP_IS_NOT_SUPPORTED
+#define WDT_OPT_PAUSE_HALTED_BY_DBG_IS_NOT_SUPPORTED
+#define WDT_DISABLE_IS_NOT_SUPPORTED
+#define WDT_FEED_CAN_STALL
+
+/* Align tests to the specific target: */
+#if defined(CONFIG_SOC_NRF54L15)
+#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
+#define WDT_WINDOW_MAX_ALLOWED (0x07CFFFFFU)
+#undef WDT_FLAG_RESET_SOC_IS_NOT_SUPPORTED
+#undef WDT_OPT_PAUSE_IN_SLEEP_IS_NOT_SUPPORTED
+#undef WDT_OPT_PAUSE_HALTED_BY_DBG_IS_NOT_SUPPORTED
+#undef WDT_DISABLE_IS_NOT_SUPPORTED
+#undef WDT_FEED_CAN_STALL
+#else
+#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
+#define WDT_WINDOW_MAX_ALLOWED (0xFFFFFFFFU)
+#endif
+
+static const struct device *const wdt = DEVICE_DT_GET(WDT_NODE);
+static struct wdt_timeout_cfg m_cfg_wdt0;
+
+/* Following variables are incremented in WDT callbacks
+ * to indicate whether interrupt was fired or not.
+ */
+volatile uint32_t m_test_06b_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_06B_TAG (0x12345678U)
+volatile uint32_t m_test_08b_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_08B_TAG (0x23456789U)
+volatile uint32_t m_test_08d_A_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_08D_A_TAG (0x3456789AU)
+volatile uint32_t m_test_08d_B_value __attribute__((section(NOINIT_SECTION)));
+#define TEST_08D_B_TAG (0x456789ABU)
+
+static void wdt_test_06b_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_06b_value = TEST_06B_TAG;
+}
+
+static void wdt_test_08b_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_08b_value = TEST_08B_TAG;
+}
+
+static void wdt_test_08d_A_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_08d_A_value = TEST_08D_A_TAG;
+}
+
+static void wdt_test_08d_B_cb(const struct device *wdt_dev, int channel_id)
+{
+	ARG_UNUSED(wdt_dev);
+	ARG_UNUSED(channel_id);
+	m_test_08d_B_value = TEST_08D_B_TAG;
+}
+
+#if !defined(WDT_DISABLE_IS_NOT_SUPPORTED)
+/**
+ * @brief wdt_disable() negative test
+ *
+ * Confirm that wdt_disable() returns
+ * -EFAULT when watchdog instance is not enabled.
+ *
+ */
+ZTEST(wdt_coverage, test_01_wdt_disable_before_wdt_setup)
+{
+	int ret;
+
+	ret = device_is_ready(wdt);
+	zassert_true(ret, "WDT device is not ready, got unexpected value of %d", ret);
+
+	/* Call wdt_disable before enabling wdt */
+	ret = wdt_disable(wdt);
+	zassert_true(ret == -EFAULT,
+		     "Calling wdt_disable before watchdog was started should return -EFAULT (-14), "
+		     "got unexpected value of %d",
+		     ret);
+}
+#else
+ZTEST(wdt_coverage, test_01_wdt_disable_before_wdt_setup)
+{
+	ztest_test_skip();
+}
+#endif
```

Comment:
With the flags as suggested above we can rewrite this without the `#ifdef`s, which improves readability IMO:

```suggestion
/**
 * @brief wdt_disable() negative test
 *
 * Confirm that wdt_disable() returns
 * -EFAULT when watchdog instance is not enabled.
 *
 */
ZTEST(wdt_coverage, test_01_wdt_disable_before_wdt_setup)
{
	int ret;

	if (!(WDT_TEST_FLAGS & WDT_DISABLE_SUPPORTED)) {
		ztest_test_skip();
	}

	ret = device_is_ready(wdt);
	zassert_true(ret, "WDT device is not ready, got unexpected value of %d", ret);

	/* Call wdt_disable before enabling wdt */
	ret = wdt_disable(wdt);
	zassert_true(ret == -EFAULT,
		     "Calling wdt_disable before watchdog was started should return -EFAULT (-14), "
		     "got unexpected value of %d",
		     ret);
}
```

Reply:
I applied changes as suggested.